Source/DafnyCore/Compilers/GoLang/Compiler-go.cs

```diff
@@ -1835,9 +1835,11 @@ protected class ClassWriter : IClassWriter {
     }
 
     protected override ConcreteSyntaxTree CreateLabeledCode(string label, bool createContinueLabel, ConcreteSyntaxTree wr) {
+      wr.WriteLine("{");
```

Comment:
Instead of explicitly adding the open and close curly braces, change the `var w = wr.Fork();` line to

    var w = wr.NewBlock(open: BlockStyle.Brace);

Reply:
I can't because we need to add a label right after the final closing brace (line 1843), unless you have another suggestion?

Reply:
I don't see that the label _after_ the closing brace is a problem, but I now see that the goto between the fork point and the closing curly is. So, the `Fork` _is_ necessary, after all. My bad.

However, when I now look at that, it seems that both the forked `w` and the `goto ...` are not indented inside the curly braces. The best way to fix that is to use a `NewBlock`.

In summary, would the following body of `CreateLabeledCode` work for you?

``` c#
var wBody = wr.NewBlock(open: BlockStyle.Brace);
var w = wBody.Fork();
var prefix = createContinueLabel ? "C" : "L";
wBody.WriteLine($"goto {prefix}{label};");
wr.Fork(-1).WriteLine($"{prefix}{label}:");
return w;
```